src/main/scala/mesosphere/marathon/metrics/current/reporters/StatsDReporter.scala

```diff
@@ -0,0 +1,156 @@
+      case v: Number => v
+    }
+
+    maybeSendAndAppend(socket, s"$name:$value|g\n")
+  }
+
+  private def reportCounter(socket: ActorRef, name: String, counter: Counter): Unit =
+    maybeSendAndAppend(socket, s"$name:${counter.getCount}|c\n")
+
+  private val histogramSnapshotSuffixes =
+    Seq("min", "mean", "p50", "p75", "p95", "p98", "p99", "p999", "max", "stddev")
+  private def reportSnapshot(socket: ActorRef, name: String, snapshot: Snapshot,
+    scaleMetrics: Boolean): Unit = {
+    val values = Seq(
+      snapshot.getMin.toDouble,
+      snapshot.getMean,
+      snapshot.getMedian,
+      snapshot.get75thPercentile(),
+      snapshot.get95thPercentile(),
+      snapshot.get98thPercentile(),
+      snapshot.get99thPercentile(),
+      snapshot.get999thPercentile(),
+      snapshot.getMax.toDouble,
+      snapshot.getStdDev)
+    val scaledValues = if (scaleMetrics) values.map(_ * durationFactor) else values
+
+    histogramSnapshotSuffixes.zip(scaledValues).foreach {
+      case (suffix, value) => maybeSendAndAppend(socket, s"$name.$suffix:$value|g\n")
+    }
+  }
+
+  private def reportHistogram(socket: ActorRef, name: String, histogram: Histogram): Unit = {
+    val count = histogram.getCount.toDouble
+    maybeSendAndAppend(socket, s"$name.count:$count|g\n")
+    reportSnapshot(socket, name, histogram.getSnapshot, false)
+  }
+
+  private val meteredSuffixes = Seq("count", "mean_rate", "m1_rate", "m5_rate", "m15_rate")
+  private def reportMetered(socket: ActorRef, name: String, meter: Metered): Unit = {
+    val values = Seq(
+      meter.getCount.toDouble,
+      meter.getMeanRate * rateFactor,
+      meter.getOneMinuteRate * rateFactor,
+      meter.getFiveMinuteRate * rateFactor,
+      meter.getFifteenMinuteRate * rateFactor)
+    meteredSuffixes.zip(values).foreach {
+      case (suffix, value) => maybeSendAndAppend(socket, s"$name.$suffix:$value|g\n")
+    }
+  }
+
+  private def reportTimer(socket: ActorRef, name: String, timer: Timer): Unit = {
+    val count = timer.getCount.toDouble
+    maybeSendAndAppend(socket, s"$name.count:$count|g\n")
+    val snapshot = timer.getSnapshot
+    reportSnapshot(socket, name, snapshot, true)
+    reportMetered(socket, name, timer)
+  }
+
+  val maxPayloadSize = 508 // the maximum safe UDP payload size
+  val buffer = new StringBuilder
+
+  private def maybeSendAndAppend(socket: ActorRef, measurement: String): Unit = {
+    require(measurement.length <= maxPayloadSize, "measurement length is too large")
+
+    if (buffer.length + measurement.length > maxPayloadSize)
+      flush(socket)
+    buffer.append(measurement)
+  }
+
+  private def flush(socket: ActorRef): Unit = {
+    if (buffer.nonEmpty) {
+      socket ! Udp.Send(ByteString(buffer.toString()), remote)
+      buffer.clear()
+    }
+  }
+}
+
+object StatsDReporter {
+  def props(metricsConf: MetricsConf, registry: MetricRegistry): Props = {
+    Props(new StatsDReporter(metricsConf, registry))
+  }
+
+}
```

Comment:
I'm concerned about all those reporters that we add. They are not rocket science but they are untested code which is also hard to test - we'll have to fiddle with ports and what not. 
Potentially, I'd even put them in their own sub-project - it might even get some stars on github since there aren't many native scala exporters for dropwizard.

Reply:
Yes,  but there is no other way for the reasons we've discussed. As for putting them into in a (sub)project, I guess, @meichstedt should chime in, because it sounds like a product/managerial decision to make.